realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,729 @@
+                } else {
+                    return table.getDate(columnIndex, rowIndex).getTime();
+                }
+            default:
+                throw new IllegalArgumentException(String.format("Column at index  %s is not a Boolean or an Integer but a %s",
+                        columnIndex, type));
+        }
+    }
+
+    /**
+     * Returns the value of the requested column as a float. Note that {@link #getType(int)} will return
+     * {@link Cursor#FIELD_TYPE_FLOAT} for both {@code float} and {@code double} columns.
+     *
+     * @param columnIndex the zero-based index of the target column.
+     * @return the value of that column as a float.
+     *
+     * @throws IllegalArgumentException if the requested column is not a float.
+     */
+    @Override
+    public float getFloat(int columnIndex) {
+        checkClosed();
+        checkPosition();
+        return table.getFloat(columnIndex, rowIndex);
+    }
+
+    /**
+     * Returns the value of the requested column as a double. Note that {@link #getType(int)} will return
+     * {@link Cursor#FIELD_TYPE_FLOAT} for both {@code float} and {@code double} columns.
+     *
+     * @param columnIndex the zero-based index of the target column.
+     * @return the value of that column as a double.
+     * @throws IllegalArgumentException if the requested column is not a double.
+     */
+    @Override
+    public double getDouble(int columnIndex) {
+        checkClosed();
+        checkPosition();
+        return table.getDouble(columnIndex, rowIndex);
+    }
+
+    /**
+     * Returns the SQLite Cursor type for a column index. Realm has more types than SQLite and some of them cannot be
+     * mapped to any meaningful SQLite type. These will return -1 and cannot be accessed using a cursor.
+     *
+     * The following types are currently not supported:
+     * <ul>
+     *   <li>{@link ColumnType#TABLE}</li>
+     *   <li>{@link ColumnType#MIXED}</li>
+     *   <li>{@link ColumnType#LINK}</li>
+     *   <li>{@link ColumnType#LINK_LIST}</li>
+     * </ul>
+     *
+     * <p>
+     * Some realm types are mapped to conform to the Cursor interface. These mappings are described below:
+     * <ol>
+     *   <li>boolean : Will return Cursor.FIELD_TYPE_INTEGER. {0 = false, 1 = true}, use getShort()/getInt()/getLong()</li>
+     *   <li>double : Will return Cursor.FIELD_TYPE_FLOAT, but must be fetched using use getDouble()</li>
+     *   <li>Date : Will return Cursor.FIELD_TYPE_INTEGER. Time in milliseconds since epoch, use getLong()</li>
+     *</ol>
+     *
+     * @param columnIndex Get the data type for this column index or -1 if the column isn't accessible.
+     * @return One of the {@code FIELD_TYPE_*}'s described in {@link Cursor}.
+     */
+    @Override
+    public int getType(int columnIndex) {
+        checkClosed();
+        ColumnType realmType = table.getColumnType(columnIndex);
+        switch (realmType) {
```

Comment:
Switch statements on enums have specific support in bytecode as well as being very readable and using less memory. A map would definitely not be faster as that would involve multiple method calls as well as memory overhead. An primitive array *might* be faster, but would also increase memory usage as we would have to keep an array around. In any case this is not a critical path method.